compiler/job.go

```diff
@@ -40,14 +40,7 @@ func NewJob(octx *op.Context, in ast.Seq, src *data.Source, head *lakeparse.Comm
 	if len(seq) == 0 {
 		return nil, errors.New("internal error: AST seq cannot be empty")
 	}
-	from, reader, err := buildFrom(seq[0], head)
-	if err != nil {
-		return nil, err
-	}
-	if from != nil {
-		seq.Prepend(from)
-	}
-	entry, err := semantic.Analyze(octx.Context, seq, src, head)
+	entry, reader, err := semantic.Analyze(octx.Context, seq, src, head)
 	if err != nil {
 		return nil, err
 	}
```

Comment:
Nit: No need to return a *kernel.Reader from semantic.Analyze since you can just do the same type assertion here.
```suggestion
	entry, err := semantic.Analyze(octx.Context, seq, src, head)
	if err != nil {
		return nil, err
	}
	reader, _ := entry[0].(*kernel.Reader)
```